src/catching_show.c

```diff
@@ -0,0 +1,327 @@
+    }
+    return FALSE;
+}
+
+BattleSetup *CatchingShow_GetBattleDTO(FieldSystem *fieldSystem) {
+    return SetupEncounter(fieldSystem, &sCatchingShow);
+}
+
+void CatchingShow_UpdateBattleResult(FieldSystem *fieldSystem, BattleSetup *setup) {
+    return UpdateBattleResultInternal(fieldSystem, setup, &sCatchingShow);
+}
+
+int CatchingShow_GetParkBallCount(FieldSystem *fieldSystem) {
+    return CATCHING_SHOW_MONS - NumMonsCaptured(&sCatchingShow);
+}
+
+int CatchingShow_CalcCatchingPoints(FieldSystem *fieldSystem) {
+    return CalcCatchingPoints(&sCatchingShow);
+}
+
+int CatchingShow_GetTimePoints(FieldSystem *fieldSystem) {
+    return GetTimePoints(&sCatchingShow);
+}
+
+u32 CatchingShow_GetTypePoints(FieldSystem *fieldSystem) {
+    return CalculateTypePoints(&sCatchingShow);
+}
+
+// Local functions
+
+static void BufferSpeciesData(u16 species, u8 *dest) {
+    GF_ASSERT(species != SPECIES_NONE && species <= NATIONAL_DEX_COUNT);
+    ReadFromNarcMemberByIdPair(dest, NARC_arc_ppark, 0, (species - 1) * 6, 6);
+}
+
+static void InitSpeciesData(FieldSystem *fieldSystem, CatchingShow *catchingShow) {
+    PalParkTransfer *transferData = Save_MigratedPokemon_Get(fieldSystem->saveData);
+    Pokemon *mon = AllocMonZeroed(HEAP_ID_4);
+    u8 narc_data[6];
+    u16 species;
+
+    for (int i = 0; i < CATCHING_SHOW_MONS; i++) {
+        catchingShow->caughtMonsOrder[i] = 0;
+        TransferDataToMon(transferData, i, mon);
+
+        species = GetMonData(mon, MON_DATA_SPECIES, NULL);
+
+        catchingShow->pokemon[i].species = species;
+        BufferSpeciesData(species, narc_data);
+
+        if (narc_data[0] != 0) {
+            catchingShow->pokemon[i].area = narc_data[PPMONDAT_OFFSET_LAND_SECTOR];
+        } else {
+            catchingShow->pokemon[i].area = narc_data[PPMONDAT_OFFSET_WATER_SECTOR] + (int)(PP_ENCTYPE_WATER_MIN - PP_ENCTYPE_LAND_MIN);
+        }
+        catchingShow->pokemon[i].rarity = narc_data[PPMONDAT_OFFSET_ENCOUTER_RATE];
+        catchingShow->pokemon[i].catchingPoints = narc_data[PPMONDAT_OFFSET_SCORE];
+        catchingShow->pokemon[i].type1 = GetMonData(mon, MON_DATA_TYPE_1, NULL);
+        catchingShow->pokemon[i].type2 = GetMonData(mon, MON_DATA_TYPE_2, NULL);
+    }
+
+    FreeToHeap(mon);
+}
+
+static int NumMonsCaptured(CatchingShow *catchingShow) {
+    int i;
+    int numMonsCaptured = 0;
+
+    for (i = 0; i < CATCHING_SHOW_MONS; i++) {
+        if (catchingShow->caughtMonsOrder[i] != 0) {
+            numMonsCaptured++;
+        }
+    }
+
+    return numMonsCaptured;
+}
+
+static void ResetStepCount(CatchingShow *catchingShow) {
+    u16 rnd = LCRandom() % 10;
+    catchingShow->steps = rnd + 5;
+}
+
+static BOOL IsStepCountZero(CatchingShow *catchingShow) {
+    if (--catchingShow->steps == 0) {
+        ResetStepCount(catchingShow);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static enum PalParkEncounterType GetEncounterArea(FieldSystem *fieldSystem, int x, int z) {
```

Comment:
Naming:
Mismatch between enum name and function name (GetEncounterArea returns an EncounterType?)